src/server/controllers.js

```diff
@@ -0,0 +1,188 @@
+const generateQuery = (endCursor, {
+	f,
+}) => {
+	let showForks = false;
+	let showSources = true;
+	let showPrivate = false;
+	let showPublic = true;
+	if (f && f.length) {
+		showForks = f.includes('forks');
+		showSources = f.includes('sources');
+		showPrivate = f.includes('private');
+		showPublic = f.includes('public');
+	}
+	return (
+		`query {
+  viewer {
+	repositories(
+	  first: 100
+	  affiliations: [OWNER, ORGANIZATION_MEMBER, COLLABORATOR]
+	  ${endCursor ? `after: "${endCursor}"` : ''}
+	  ${showForks === showSources ? '' : showForks ? 'isFork: true' : 'isFork: false'}
+	  ${showPrivate === showPublic ? '' : showPublic ? 'privacy: PUBLIC' : 'privacy: PRIVATE'}
+	) {
+	  totalCount
+	  pageInfo {
+		endCursor
+		hasNextPage
+	}
+	nodes {
+		name
+		nameWithOwner
+		defaultBranchRef {
+			name
+			branchProtectionRule {
+				allowsForcePushes
+				allowsDeletions
+				dismissesStaleReviews
+				requiredApprovingReviewCount
+				requiresApprovingReviews
+				requiresCodeOwnerReviews
+				requiresConversationResolution
+				restrictsPushes
+			}
+		}
+		deleteBranchOnMerge
+		hasIssuesEnabled
+		hasProjectsEnabled
+		hasWikiEnabled
+		forkingAllowed
+		isArchived
+		autoMergeAllowed
+		isBlankIssuesEnabled
+		isFork
+		isPrivate
+		isSecurityPolicyEnabled
+		isTemplate
+		licenseInfo {
+			name
+		}
+		mergeCommitAllowed
+		owner {
+			login
+		}
+		rebaseMergeAllowed
+		squashMergeAllowed
+		createdAt
+		updatedAt
+		pushedAt
+		viewerHasStarred
+		viewerPermission
+		viewerSubscription
+	  }
+	}
+  }
+  rateLimit {
+	cost
+	remaining
+  }
+}
+`);
+};
+
+/*
+ * DOUBT: Copy-Pasted this function here as I was having issues with importing the function
+ * from src/commands/detail.js
+ */
+const detail = async (flags) => {
+	if (flags.m) {
+		return listMetrics(getMetrics(metricNames));
+	}
+	let metrics;
+	if (flags.p?.length > 0) {
+		metrics = getMetrics([
+			'Repository',
+			'isFork',
+			'isPrivate',
+			...metricNames.filter((name) => flags.p.includes(name)),
+		]);
+	} else {
+		metrics = getMetrics(metricNames);
+	}
+
+	// Additional Filter on repos
+	let filter;
+	if (flags.f?.length === 1 && flags.f[0] === 'templates') {
+		filter = (repo) => repo.isTemplate;
+	}
+	// Get all repositories
+	const { repositories } = await getRepositories(generateQuery, flags, filter);
+	if (!flags.s) {
+		repositories.sort((a, b) => new Date(b.updatedAt) - new Date(a.updatedAt));
+	}
+	// Generate output table
+	const table = generateDetailTable(metrics, repositories, {
+		actual: flags.actual, all: flags.all, goodness: flags.goodness, sort: flags.s, unactionable: flags.unactionable,
+	});
+
+	if (table) {
+		if (flags.serve) {
+			return table;
+		}
+	}
+	return null;
+};
+
+const executeCommand = async (req, res) => {
+	const { command } = req.body;
+	const argv = parse(command);
+	argv.serve = true;
+	argv.goodness = true;
+	// DOUBT: Is there perhaps a better way or a better place to fetch the token?
+	argv.token = process.env.GH_TOKEN;
```

Comment:
the CLI should already be populating this - perhaps that logic should be shared in a module?

Reply:
Printing the argv just before the detail command is called prints this:
**code:**
<img width="349" alt="Screenshot 2021-12-28 at 11 28 02 PM" src="https://user-images.githubusercontent.com/52193408/147593879-1c29994f-5085-4b07-9f37-da99715f28c9.png">

**command:**
<img width="376" alt="Screenshot 2021-12-28 at 11 28 41 PM" src="https://user-images.githubusercontent.com/52193408/147593915-380f9b54-4850-47b6-8d23-f9db74766c3f.png">

**console:**
<img width="332" alt="Screenshot 2021-12-28 at 11 29 12 PM" src="https://user-images.githubusercontent.com/52193408/147593945-d07abe7d-5ab3-40f9-85d7-69045cff3b64.png">

Reply:
In other words, since I am using `yargs-parser` to programmatically generate the argvs, I need to pass the token programmatically too, so I need to pass the token here itself.

Reply:
redundant conversation thread, continued in https://github.com/ljharb/repo-report/pull/59#discussion_r776534542